src/v/cluster/commands.h

```diff
@@ -137,6 +138,12 @@ using move_partition_replicas_cmd = controller_command<
   model::record_batch_type::topic_management_cmd,
   serde_opts::adl_and_serde>;
 
+using move_topic_replicas_cmd = controller_command<
+  model::topic_namespace,
+  std::vector<move_topic_replicas_data>,
+  move_topic_replicas_cmd_type,
+  model::record_batch_type::topic_management_cmd>;
```

Comment:
does this need a serde_opts::serde_only tag? (cc @andrwng)

Reply:
The default is `serde_opts::serde_only`; feel free to make that more obvious by declaring it explicitly.